crate/cli/src/commands.rs

```diff
@@ -87,7 +87,7 @@ impl CoreFindexActions {
     /// # Errors
     /// - If the configuration file is not found or invalid
     #[allow(clippy::unit_arg)] // println! does return () but it prints the output of action.run() beforehand, nothing is "lost" and hence this lint will only cause useless boilerplate code
-    pub async fn run(&self, findex_client: &mut FindexRestClient) -> CliResult<()> {
+    pub async fn run(&self, mut findex_client: FindexRestClient) -> CliResult<()> {
```

Comment:
Do you really need this `mut` ? Can't you pass ownership of the config where it is needed instead ?